lib/plugins/aws/package/compile/events/schedule.js

```diff
@@ -136,9 +136,13 @@ class AwsCompileScheduledEvents {
               Name = event.schedule.name;
               timezone = event.schedule.timezone;
               Description = event.schedule.description;
-              roleArn = {
-                'Fn::GetAtt': ['IamRoleLambdaExecution', 'Arn'],
-              };
+
+              const functionLogicalId = this.provider.naming.getLambdaLogicalId(functionName);
+              const functionResource = resources[functionLogicalId];
+
+              roleArn = resources.IamRoleLambdaExecution
+                ? { 'Fn::GetAtt': ['IamRoleLambdaExecution', 'Arn'] }
+                : functionResource.Properties.Role;
```

Comment:
I was thinking that if there was another function that uses the default execution role, then the role will be created and we can use it here.

Reply:
But here we're just referencing some external role, and creating one, right? Shouldn't we then always reference same role to which function is attached (?)

Reply:
Let's say we have 2 functions. 1 uses a custom role and the 1 does not have role defined and will use the default execution role. Then we want to use the default execution role for all scheduler schedules because it will be created have the right permissions. 

If both functions use the custom role, then the default execution role will not be created. Then the scheduler will then use the same role as the function and the warning will be displayed.

The point is to always use the default execution role if it's being created.

Reply:
What is the downside of always using the role to which function is attached? We anyway create different scheduler resource for each configured event

Reply:
It indeed may feel more convenient to simply attach to role which may not require further customisations.

Yet logically I think event, with no exceptions, should work in context of a role which is assigned to function.
It can be assumed as security breach if suddenly we rely on role that's dedicated for function A, when configuring the resource for function B.

Many users would prefer to have IAM permissions isolated and configured per function (see https://github.com/serverless/serverless/issues/4313) So now attempting here to reach out of role that's used for function doesn't feel as right direction

Reply:
I will change it back to what I had before, to use the same role as the function. 

However, it's my opinion that the scheduler role should not be the same as the Lambda execution role and should be customizable just like for functions. But we can save that for a future issue.